server/src/main.rs

```diff
@@ -135,40 +134,46 @@ async fn run_http() -> anyhow::Result<()> {
 pub fn configure_routes(cfg: &mut web::ServiceConfig) {
     let generated = generate();
 
-    // Base path "{url}/api/v1"
-    // POST "/query" ==> Get results of the SQL query passed in request body
-    cfg.service(web::resource(query_path()).route(web::post().to(handlers::event::query)))
-        .service(
-            // logstream API
-            web::resource(logstream_path("{logstream}"))
-                // PUT "/logstream/{logstream}" ==> Create log stream
-                .route(web::put().to(handlers::logstream::put))
-                // POST "/logstream/{logstream}" ==> Post logs to given log stream
-                .route(web::post().to(handlers::event::post_event))
-                // DELETE "/logstream/{logstream}" ==> Delete log stream
-                .route(web::delete().to(handlers::logstream::delete))
-                .app_data(web::JsonConfig::default().limit(MAX_EVENT_PAYLOAD_SIZE)),
-        )
-        .service(
-            web::resource(alert_path("{logstream}"))
-                // PUT "/logstream/{logstream}/alert" ==> Set alert for given log stream
-                .route(web::put().to(handlers::logstream::put_alert))
-                // GET "/logstream/{logstream}/alert" ==> Get alert for given log stream
-                .route(web::get().to(handlers::logstream::get_alert)),
-        )
-        // GET "/logstream" ==> Get list of all Log Streams on the server
-        .service(web::resource(logstream_path("")).route(web::get().to(handlers::logstream::list)))
-        .service(
-            // GET "/logstream/{logstream}/schema" ==> Get schema for given log stream
-            web::resource(schema_path("{logstream}"))
-                .route(web::get().to(handlers::logstream::schema)),
-        )
-        // GET "/liveness" ==> Livenss check as per https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-command
-        .service(web::resource(liveness_path()).route(web::get().to(handlers::liveness)))
-        // GET "/readiness" ==> Readiness check as per https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-readiness-probes
-        .service(web::resource(readiness_path()).route(web::get().to(handlers::readiness)))
-        // GET "/" ==> Serve the static frontend directory
-        .service(ResourceFiles::new("/", generated));
+    cfg.service(
+        // Base path "{url}/api/v1"
+        web::scope(&base_path())
```

Comment:
@nitisht base_path is used in this scope thus every path grouped underneath it will share same base url i.e /api/v1. I changed the related functions to not return full path.

Reply:
ah okay..we need to test all the APIs then, just to make sure nothing is broken

Reply:
Yep